javascript/modules/client/src/client.ts

```diff
@@ -282,32 +308,124 @@ export class Client extends ClientCore implements IClient {
       override.value = amount;
     }
 
-    return daoInstance.estimateGas.deposit(
-      tokenAddress,
-      amount,
-      reference,
-      override,
-    ).then((gasLimit) => {
-      return this.web3.getApproximateGasFee(gasLimit.toBigInt());
-    });
+    return daoInstance.estimateGas
+      .deposit(tokenAddress, amount, reference, override)
+      .then(gasLimit => {
+        return this.web3.getApproximateGasFee(gasLimit.toBigInt());
+      });
+  }
+
+  private async _getDaoBalances(
+    daoIdentifier: string
+  ): Promise<TokenBalance[]> {
+    // TODO: Implement actual fetch logic using subgraph.
+    // Note: it would be nice if the client could be instantiated with dao identifier
+
+    if (!daoIdentifier) {
+      throw new Error("A DAO identifier is needed");
+    }
+
+    const tokenBalances: TokenBalance[] = tokenList.map(
+      (token: TokenBalance["token"]) => ({
+        id: `${daoIdentifier}_${token.id}`,
+        token,
+        // Generate a random balance amount between [0, 1000]
+        balance: BigInt(
+          (Math.floor(Math.random() * (1000 - 1 + 1) + 1) * 10) ** 18
+        ),
+        // Generate a random date in the past
+        lastUpdated: Math.floor(
```

Comment:
I would return a native JS `Date` object. 

Also, I would name it `lastUpdate`

Reply:
On the web app, we decided to manage everything based on timestamp, (according to the graph's current structure). I like this idea but on web app we're mostly using date-fns to show a human-centered format. the timestamp would be better for us at this moment